src/main/java/com/duke/command/CommandMark.java

```diff
@@ -0,0 +1,77 @@
+package com.duke.command;
+
+import java.io.IOException;
+
+import com.duke.exception.DukeEmptyListException;
+import com.duke.exception.DukeInvalidArgumentException;
+import com.duke.modules.Storage;
+import com.duke.modules.TaskList;
+import com.duke.tasks.Task;
+
+/**
+ * Represents a chatbot command for marking Tasks in the TaskList
+ */
+public class CommandMark extends Command {
+    private String input;
+    private TaskList taskList;
+
+    /**
+     * Constructor for this class.
+     *
+     * @param input    The string of arguments entered by the user, excluding the command word.
+     * @param taskList The tasklist meant for the Tasks to be added to.
+     */
+    public CommandMark(String input, TaskList taskList) {
+        super();
+        this.input = input;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Returns a string of the result of executing the intended function of this class.
+     * This string is wrapped in a CommandResult object
+     *
+     * @return A CommandResult object containing the result message.
+     */
+    @Override
+    public CommandResult execute() {
+        try {
+            return new CommandResult(taskMarker(input));
+        } catch (DukeInvalidArgumentException | DukeEmptyListException e) {
+            return new CommandResult(e.getMessage());
+        } catch (NumberFormatException e) {
+            return new CommandResult(
+                    "Valid numerical number must be given:" + "\nEg: mark 12");
+        } catch (IndexOutOfBoundsException e) {
+            return new CommandResult(
+                    "Invalid task chosen, please ensure that the number given is correct");
+        } catch (IOException e) {
+            return new CommandResult("Unable to save list."
+                    + "Please check if you have permission to write to files in the following directory: "
+                    + Storage.getInstance().getDirectoryPath());
+        }
+    }
+

```

Comment:
thorough checking of different possible exceptions, each returning different prompts, is a nice touch 👍